plugins/com.google.cloud.tools.eclipse.appengine.newproject.maven/src/com/google/cloud/tools/eclipse/appengine/newproject/maven/MavenAppEngineStandardWizardPage.java

```diff
@@ -226,17 +228,32 @@ protected void checkFlipToNext() {
    *
    * @return true if valid, false if there is a problem
    */
-  public boolean validatePage() {
+  private boolean validatePage() {
     setMessage(null);
     setErrorMessage(null);
 
     // order here should match order of the UI fields
 
-    String location = locationField.getText().trim();
-    if (!useDefaults() && location.isEmpty()) {
-      setMessage(Messages.getString("PROVIDE_LOCATION"), INFORMATION); //$NON-NLS-1$
-      return false;
-    }
+    if (!useDefaults()) {
+      String location = locationField.getText().trim();
+      if (location.isEmpty()) {
+	    setMessage(Messages.getString("PROVIDE_LOCATION"), INFORMATION); //$NON-NLS-1$
+	    return false;
+      } else {
+          java.nio.file.Path path = Paths.get(location);
+    	  if (Files.exists(path) && !Files.isDirectory(path)) {
+            String message = MessageFormat.format(Messages.getString("FILE_LOCATION"), location); //$NON-NLS-1$
+    	  	setMessage(message, INFORMATION);
+            return false;    		  
+          } else if (Files.exists(path) && !Files.isWritable(path)) {
+            String message = MessageFormat.format(Messages.getString("NONWRITABLE"), location); //$NON-NLS-1$
+      	  	setMessage(message, INFORMATION);
+            return false;         	  
+          }
+    	  // TODO how to check if a directory that doesn't exist could be created?
```

Comment:
It's a little more complex than that. We have to walk up the tree to find out whether the first directory that exists is writable or not.